### DISCUSSION TITLE
SPICE parameters for waveguides

### DISCUSSION THREAD

**Discussion by jevillegasd** [SiEPIC/SiEPIC_EBeam_PDK#249] [Ideas]
Hey Everyone!

I was wondering how are the standards SiEPIC uses to define SPICE parameter labels. I can see in the different cells in the library that they are directly declared while inserting the text label starting with 'Spice_param:', but does the parameter name follow a specific format or convention, or does it just depends on the compact model used?

Would it make sense to include in the waveguides.xml specification a set of optional SPICE parameters labels to allow compatibility with CMLs that don't follow the same structure? This would help only for the waveguides, but if I am not mistaken wg are the only components that are mostly defined in the tools instead of the PDKs. 

The XML could have something like
<SPICE_labels>
   <wg_width> ancho </wg_width>
   <wg_length> longitud </wg_length>
</SPICE_labels>

Specifically,  I am currently using a CML library from a foundry whose model for a waveguide in Interconnect uses 'width' instead of 'wg_width' as a variable name. Right now the only way I can connect the waveguide in SiEPIC with this model would be to create my own path_to_waveguide() function (or change the one in the core) or maybe create a model in Interconnect that uses the other foundry's model and inside associate the correct variable names... all options sound rather messy.

Any thoughts?

Warm regards, Juan

**Comment by jevillegasd** [comment]
I managed to do the trick of modifying the text after the cell is created. I still think its rather dirty but it may help someone facing a similar situation. This is the code I used in the waveguide produce method:

```
def produce_impl(self):
    from SiEPIC.utils import     get_layout_variables
    from SiEPIC.utils.layout import layout_waveguide4
    TECHNOLOGY, lv, ly, top_cell = get_layout_variables()
    LayerDevRecN  = self.layout.layer(TECHNOLOGY['DevRec']) 
    self.layout.dbu = ly.dbu
    self.waveguide_length = layout_waveguide4(self.cell, self.path, self.waveguide_type)
    # Look for things on DevRec
    iter_sh = self.cell.begin_shapes_rec(LayerDevRecN)
    while not(iter_sh.at_end()):
        if iter_sh.shape().is_text():
          shape = iter_sh.shape();
          text = shape.text
          if text.string.find("Spice_param:") > -1:
              newTsring =  text.string.replace('wg_width', 'width', 1);
              new_text = pya.Text(newTsring, shape.text_trans,shape.text_size ,-1);
              new_text.halign = text.halign
              shape.text = new_text
        iter_sh.next()
```

**Comment by lukasc-ubc** [comment]
Hi @jevillegasd 

I'm sorry I missed your post!  I was sailing at the time and missed the notification.

Is this something you would like to discuss further?

regards,
Lukas

**Reply by jevillegasd** [reply]
Hey @lukasc-ubc !

It’s always a good excuse to be sailing! The workaround I mentioned works pretty well. The issue it has is that it needs to change the waveguide produce method from what would normally be in a PDK. I wonder if adding an optional “parameter map” in the waveguide.xml specification could be used instead. something like :
```
<cml_map>
   <wg_width>width </wg_width>
   …
</cml_map>

```

Juan

**Reply by jevillegasd** [reply]
Ok I just saw I’m suggesting the same thing as my first comment 😅. 

I can give it a try and push something on how that would like if it’s a feature that may be useful.
